src/common.c

```diff
@@ -1025,15 +1028,22 @@ size_t ofi_byteq_readv(struct ofi_byteq *byteq, struct iovec *iov,
 }
 
 void ofi_byteq_writev(struct ofi_byteq *byteq, const struct iovec *iov,
-		      size_t cnt, size_t offset)
+		      size_t cnt)
 {
-	size_t len;
+	size_t i;
+
+	assert(ofi_total_iov_len(iov, cnt) <= ofi_byteq_writeable(byteq));
+
+	if (cnt == 1) {
```

Comment:
I think at this level of abstraction, it isn't a problem, but I was thinking about if this optimization could affect ordering by skipping data that may be sitting in the byteq